command.go

```diff
@@ -225,8 +225,11 @@ func min(a int, b int) int {
 // Close method wrapper
 // commandWriter implements io.Closer interface
 func (w *commandWriter) Close() error {
+	if w.eof {
+		return errors.New("commandWriter already closed")
+	}
 	w.eof = true
-	return w.Close()
+	return w.sendInput(nil)
```

Comment:
Shouldn't it be `sendInput(nil)` then `Close()`?
Otherwise the command will never be closed.

Reply:
I don't think so since `Close()` here would be a recursive call, which wouldn't do anything except return an error (since `eof` is always true here), but perhaps I'm missing something.

Reply:
Sorry I meant `w.Close()` in addition to `w.sendInput(nil)`.

This isn't really a recursive call, and `Command.Close()` will signal the server that the command is now finished by sending the `rsp:Signal` command (because I'm not sure finishing the input stream is enough to deallocate everything server side and make the shell ready for another command).

I need to perform some tests to see what is the behavior with your modifications.

Reply:
Sorry for the delay in responding!

After looking at `rsp:Signal` in the protocol spec, I definitely agree that `sendInput(nil)` should be followed by `Command.Close()`, but I think that should happen after the `wg.Wait()` in `Client.Run()`, `Client.RunWithString()`, and `Client.RunWithInput()` to allow the remote shell to finish sending output. Does that sound right to you?

Reply:
@nwt, 

Yes this is probably a more correct solution.
Can you implement it and see if that still works correctly (including the tests)?

Thanks!